### PR TITLE
Added 4 more snippets (see description)

### PR DESCRIPTION
**Pull request by soumitradev**
**Intermediate level:**
intersection_by.md: Gets intersection of 2 lists after applying a function on each of their elements. Adapted from: https://github.com/30-seconds/30-seconds-of-python/blob/master/snippets/intersection_by.md

longest.md: Gets element with maximum `length()` value. Adapted from: https://github.com/30-seconds/30-seconds-of-python/blob/master/snippets/longest_item.md

root_mean_square.md: Gets the root mean square of arguments passed. See https://en.wikipedia.org/wiki/Root_mean_square

**Advanced level:**
Added an Advanced snippet luhn.md that checks if a number is luhn-valid or not. See https://en.wikipedia.org/wiki/Luhn_algorithm and https://github.com/30-seconds/30-seconds-of-code/blob/ebbb22db7cddbaac7f33c5432fe5600b2e78df42/snippets/luhnCheck.md

### CODE REVIEW

**Comment by soumitradev** [comment]
Just removed [WIP] from title.

**Comment by soumitradev** [comment]
Added an Advanced snippet luhn.md that checks if a number is luhn-valid or not. See https://en.wikipedia.org/wiki/Luhn_algorithm

File: snippets/luhn.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: luhn
+tags: math,utility,advanced
+---
+
+Implementation of the [Luhn Algorithm](https://en.wikipedia.org/wiki/Luhn_algorithm) used to validate a variety of identification numbers, such as credit card numbers, IMEI numbers, National Provider Identifier numbers, etc.
+
+Define a `digit_sum()` method to get the sum of the digits of a number. Then, first format the number to be luhn-checked as a String, and save the last digit using `string(num[end])` as the check digit.
+Then, get the sum of the alternate digits of the reversed number as `parta` using slicing, and `parse` on each digit as to get an `Int` from a `String`, and using `digit_sum()` to get the sum of the digits.
+Then, get the digits of the reversed number alternately (this time the other numbers, not the ones we just summed up) using `reverse()`, `parse()`, `digits()`, and slicing, multiply them by two, get each of their sums of digits using `digit_sum.()`, and save the sum of the sums of digits as `partb` using `sum()`.
```
Comment (logankilpatrick) [inline snippets/luhn.md]:
```sugestion
After that, get the digits of the reversed number alternately (this time the other numbers, not the ones we just summed up) using `reverse()`, `parse()`, `digits()`, and slicing, multiply them by two, get each of their sums of digits using `digit_sum.()`, and save the sum of the sums of digits as `partb` using `sum()`.
```

Comment (logankilpatrick) [reply snippets/luhn.md]: Replace "then" with "After that,"

**Pull request merged**